app/src/main/java/org/fossasia/openevent/app/core/ticket/detail/MyPrintDocumentAdapter.java

```diff
@@ -0,0 +1,137 @@
+package org.fossasia.openevent.app.core.ticket.detail;
+
+import android.content.Context;
+import android.graphics.Canvas;
+import android.graphics.Color;
+import android.graphics.Paint;
+import android.graphics.pdf.PdfDocument;
+import android.os.Build;
+import android.os.Bundle;
+import android.os.CancellationSignal;
+import android.os.ParcelFileDescriptor;
+import android.print.PageRange;
+import android.print.PrintAttributes;
+import android.print.PrintDocumentAdapter;
+import android.print.PrintDocumentInfo;
+import android.print.pdf.PrintedPdfDocument;
+import android.support.annotation.RequiresApi;
+
+import org.fossasia.openevent.app.data.ticket.Ticket;
+import org.fossasia.openevent.app.utils.DateUtils;
+
+import java.io.FileOutputStream;
+import java.io.IOException;
+
+@RequiresApi(api = Build.VERSION_CODES.KITKAT)
+public class MyPrintDocumentAdapter extends PrintDocumentAdapter {
+
+    Context context;
+    private int pageHeight;
+    private int pageWidth;
+    private PdfDocument myPdfDocument;
+    private final int TOTAL_PAGES = 1;
+    private Ticket ticket;
+
+    public MyPrintDocumentAdapter(Context context, Ticket ticket) {
+        this.context = context;
+        this.ticket = ticket;
+    }
+
+    @Override
+    public void onLayout(PrintAttributes oldAttributes,
+                         PrintAttributes newAttributes,
+                         CancellationSignal cancellationSignal,
+                         LayoutResultCallback callback, Bundle extras) {
+        myPdfDocument = new PrintedPdfDocument(context, newAttributes);
+
+        pageHeight =
+            newAttributes.getMediaSize().getHeightMils() / 1000 * 72;
+        pageWidth =
+            newAttributes.getMediaSize().getWidthMils() / 1000 * 72;
```

Comment:
Prev line

Reply:
What's the logic to /1000*72

Reply:
The dimensions are stored in the object in the form of thousandths of an inch. Since the methods that will use these values work in units of 1/72 of an inch these numbers are converted before they are stored.